pilot/cmd/pilot-agent/status/ready/listeners.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2018 Istio Authors
```

Comment:
Can we keep the logic in `istio.io/pilot/pkg`? Maybe in the proxy directory?

Reply:
I had it there originally, but there was a concern about proxy-agent pulling in too many dependencies.  I suppose so long as it's in its own package it should be fine.  Do you have a preference of location?